src/client/game.cpp

```diff
@@ -2195,44 +2195,44 @@ void Game::toggleMinimap(bool shift_pressed)
 
 	u32 hud_flags = client->getEnv().getLocalPlayer()->hud_flags;
 
-	MinimapMode mode = MINIMAP_MODE_OFF;
-	if (hud_flags & HUD_FLAG_MINIMAP_VISIBLE) {
-		mode = mapper->getMinimapMode();
-		mode = (MinimapMode)((int)mode + 1);
-		// If radar is disabled and in, or switching to, radar mode
-		if (!(hud_flags & HUD_FLAG_MINIMAP_RADAR_VISIBLE) && mode > 3)
-			mode = MINIMAP_MODE_OFF;
-	}
+	MinimapMode mode = mapper->getMinimapMode();
+	mode = static_cast<MinimapMode>(static_cast<int>(mode) + 1);
+
+	// If radar is disabled and in or while switching to radar mode, set to OFF
+	if (!(hud_flags & HUD_FLAG_MINIMAP_RADAR_VISIBLE) &&
+			mode > MINIMAP_MODE_SURFACEx4)
+		mode = MINIMAP_MODE_OFF;
 
-	m_game_ui->m_flags.show_minimap = true;
+	std::string status_text;
 	switch (mode) {
 		case MINIMAP_MODE_SURFACEx1:
-			m_game_ui->showTranslatedStatusText("Minimap in surface mode, Zoom x1");
+			status_text = "Minimap in surface mode, Zoom x1";
```

Comment:
This must be wrapped into a function which is contained in the gettext parser script since it does not compile the source files. Otherwise these sentences will disappear from the translation files, which surely not intended.

Reply:
Thanks for letting me know, this wasn't my intention of course. :)

Reply:
@SmallJoker What solution would you suggest with respect to https://github.com/minetest/minetest/pull/8441/files#diff-7f71a92de20504e8a5636b4317ea67b2R2233?